pkg/gen/typegen.go

```diff
@@ -530,14 +560,189 @@ func makePythonType(resourceType, propName string, prop map[string]interface{},
+				elemType := makeDotnetType(
+					resourceType, propName, prop["items"].(map[string]interface{}), opts)
+				return fmt.Sprintf("%s[]>", elemType[:len(elemType)-1])
+			case outputsAPI:
+				elemType := makeDotnetType(
+					resourceType, propName, prop["items"].(map[string]interface{}), opts)
+				return fmt.Sprintf("ImmutableArray<%s>", elemType)
+			case inputsAPI:
+				elemOpts := opts
+				elemOpts.forceNoWrap = true
+				elemType := makeDotnetType(
+					resourceType, propName, prop["items"].(map[string]interface{}), elemOpts)
+				return fmt.Sprintf("InputList<%s>", elemType)
+			}
+		} else if tstr == "integer" {
+			return wrapType("int")
+		} else if tstr == "boolean" {
+			return wrapType("bool")
+		} else if tstr == "number" {
+			return wrapType("double")
+		} else if tstr == "object" {
+			vtype := "string"
+			if additionalProperties, exists := prop["additionalProperties"]; exists {
+				switch opts.generatorType {
+				case provider:
+					elemOpts := opts
+					elemOpts.forceNoWrap = true
+					vtype = makeDotnetType(
+						resourceType, propName, additionalProperties.(map[string]interface{}), elemOpts)
+				case outputsAPI:
+					vtype = makeDotnetType(
+						resourceType, propName, additionalProperties.(map[string]interface{}), opts)
+				case inputsAPI:
+					elemOpts := opts
+					elemOpts.forceNoWrap = true
+					vtype = makeDotnetType(
+						resourceType, propName, additionalProperties.(map[string]interface{}), elemOpts)
+				}
+			}
+			switch opts.generatorType {
+			case inputsAPI:
+				return fmt.Sprintf("InputMap<%s>", vtype)
+			case outputsAPI:
+				return fmt.Sprintf("ImmutableDictionary<string, %s>", vtype)
+			case provider:
+				return fmt.Sprintf("Output<ImmutableDictionary<string, %s>>", vtype)
+			}
+		} else if tstr == "string" && resourceType == "io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta" && propName == "namespace" {
+			// Special case: `.metadata.namespace` should either take a string or a namespace object
+			// itself.
+
+			switch opts.generatorType {
+			case inputsAPI:
+				// TODO: Enable metadata to take explicit namespaces, like:
+				// return "pulumi.Input<string> | Namespace"
+				return "Input<string>"
+			case outputsAPI:
+				return "string"
+			}
+		}
+
+		return wrapType(tstr)
+	}
+
+	ref := stripPrefix(prop["$ref"].(string))
+	var argsSuffix string
+	var stringArr string
+	switch opts.generatorType {
+	case inputsAPI:
+		argsSuffix = "Args"
+		stringArr = "InputList<string>"
+	case outputsAPI:
+		argsSuffix = ""
+		stringArr = "ImmutableArray<string>"
+	case provider:
+		argsSuffix = ""
+		stringArr = "string[]"
+	default:
+		panic(fmt.Sprintf("unrecognized generator type %d", opts.generatorType))
+	}
+
+	isSimpleRef := true
+	switch ref {
+	case quantity:
+		ref = "string"
+	case intOrString:
+		return oneOf("int", "string")
+	case v1Fields, v1FieldsV1, rawExtension:
+		// TODO: This is quite possibly wrong - these are actually JSON objects
+		ref = "string"
```

Comment:
@CyrusNajmabadi This and a little further below are two places where the type really should be `object` - but we can't represent this.  The JSON Schema for these allows any JSON object whatsoever.  Is it possible to represent this in our .NET projection?

Reply:
Can we json stringify this so it is a string in the .net side?  Alternatively, we could define this in .net as an actual Json type and the marshal properly to that.

Reply:
Opened #889 to track this - and going to go ahead and get this PR merged without this first (since it already works for most resources).  Will add your note over on that issue.